src/components/templates/org/signup/components/step-form-factory.tsx

```diff
@@ -0,0 +1,150 @@
+import useTranslation from 'next-translate/useTranslation';
+import dynamic from 'next/dynamic';
+import { useEffect } from 'react';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useLocalStorage } from '@solana/wallet-adapter-react';
+import {
+  FormProvider,
+  useForm,
+  Resolver,
+  Path,
+  ValidationMode,
+} from 'react-hook-form';
+
+import { InputAdornment, TextField } from '@mui/material';
+
+import { categories } from '../../../../../constants/dao';
+import { localStorageKeys } from '../../../../../constants/local-storage-keys';
+import StepFormHeader from './step-form-header';
+
+const CategoriesInput = dynamic(
+  () => {
+    return import('../../../../molecules/categories-input');
+  },
+  { ssr: false }
+);
+
+type InputProps<T> = {
+  required?: boolean;
+  name: Path<T>;
+  type: string;
+  helperText?: string;
+  multiline?: boolean;
+  initialValue?: any;
+  startAdornment?: string;
+};
+
+type StepFormProps<T> = {
+  handleStep: (newValue: boolean) => void;
+  input: InputProps<T>;
+  schema: any;
+  updateFormState: (newValue: any) => void;
+  validationMode?: keyof ValidationMode;
+};
+
+export default function StepFormFactory<T>({
+  handleStep,
+  schema,
+  input,
+  updateFormState,
+  validationMode = 'all',
+}: StepFormProps<T>) {
+  const { t } = useTranslation('org-signup');
+
+  const methods = useForm<T>({
+    resolver: yupResolver(schema) as Resolver<T, object>,
+    mode: validationMode,
+  });
+
+  const { register, formState, watch, setValue, trigger } = methods;
+  const { isValid } = formState;
+  const [formValue, updateFormValue] = useLocalStorage(
+    localStorageKeys.org_signup,
+    null
+  );
+
+  useEffect(() => {
+    if (formValue && formValue[input.name]) {
+      setValue(input.name, formValue[input.name]);
+      trigger(input.name);
+    }
+  }, []);
```

Comment:
Instead of using useEffect to set default values, you could get the values of local state before the definition of the `useForm` and use the `defaultValue` property


```suggestion
const [formValue, updateFormValue] = useLocalStorage(
    localStorageKeys.org_signup,
    null
  );
  const methods = useForm<T>({
    resolver: yupResolver(schema) as Resolver<T, object>,
    mode: validationMode,
    defaultValues: formValue[input.name]
  });

  const { register, formState, watch, setValue, trigger } = methods;
  const { isValid } = formState;

```